### PR TITLE
fix: Enforce revision ID and model names for future contributions

### PR DESCRIPTION
**Pull request by KennethEnevoldsen**
Added tests to:
- Enforce revision ID for new models
- Enforce consistent model names for new models

Added this change mainly due to duplicates models results and due to recent addition not including a model revision (which I we should probably discourage).

This fixes model results for:
- voyage

(needed for the leaderboard)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,6 +4,10 @@ type: evaluation
 submission_name: MTEB
 ---
 
+> [!NOTE]  
+> Previously it was possible to submit models results to MTEB by adding the results to the model metadata. This is no longer an option as we want to ensure high quality metadata.
```
Comment (isaac-chung) [inline README.md:8]: Question: is that currently enforced? Or will only be in effect once the new leaderboard goes public?

File: tests/test_ensure_correct_metadata.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            "",
+            "na",
+            "no-revision-available",
+        ]
+
+
+@pytest.mark.parametrize("model_rev_pair", model_rev_pairs)
+def test_organization_is_specified_for_new_additions(model_rev_pair):
+    """
+    Models added after 26th of November should include a organization ID within their name, e.g. "myorg/my_embedding_model".
```
Comment (isaac-chung) [inline tests/test_ensure_correct_metadata.py:65]:
```suggestion
    Models added after November 26, 2024 should include a organization ID within their name, e.g. "myorg/my_embedding_model".
```

File: tests/test_ensure_correct_metadata.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    if model_folder.name not in [".DS_Store"]
+    and rev_folder.name not in [".DS_Store"]
+    and ((model_folder.name, rev_folder.name) not in folders_without_meta)
+]
+
+
+@pytest.mark.parametrize("model_rev_pair", model_rev_pairs)
+def test_model_meta_in_folders(model_rev_pair):
+    """
+    Models added after the 26th of November should contain a model_meta.json file
```
Comment (isaac-chung) [inline tests/test_ensure_correct_metadata.py:20]:
```suggestion
    Models added after November 26, 2024 should contain a model_meta.json file
```

**Review by isaac-chung — APPROVED** [review]
Looks good. Some non-blocking Qs and suggestions.

File: README.md (hunk 1 above)
Comment (KennethEnevoldsen) [reply README.md:8]: The new leaderboard does not fetch it automatically (only fetches from here)

However, we had a PR that fetches everything from HF (once, but probably not something we will run again).

E.g. the models:

 'vectoriseai/bge-base-en-v1.5',
 'vectoriseai/bge-large-en-v1.5',
 'vectoriseai/bge-small-en',
 'vectoriseai/bge-small-en-v1.5',
 'vectoriseai/e5-base',
 'vectoriseai/e5-base-v2',
 'vectoriseai/e5-large',
 'vectoriseai/e5-large-v2',
 'vectoriseai/e5-small-v2',
 'vectoriseai/gte-base',
 'vectoriseai/gte-large',
 'vectoriseai/gte-small',
 'vectoriseai/multilingual-e5-large',

are all copies and then creates duplicates on the leaderboard.

Comment (KennethEnevoldsen) [reply README.md:8]: pr. #57 removes these

**Pull request merged**